audresample/core/lib.py

```diff
@@ -10,13 +10,23 @@
 bin_path = os.path.join(root, 'bin')
 if platform.system() == 'Windows':  # pragma: no cover
     # for some reason we have to load soxr.dll first...
-    lib_path = os.path.join(bin_path, 'soxr.dll')
+    lib_path = os.path.join(bin_path, 'windows', 'soxr.dll')
     ctypes.cdll.LoadLibrary(lib_path)
-    lib_path = os.path.join(bin_path, 'audresample.dll')
+    lib_path = os.path.join(bin_path, 'windows', 'audresample.dll')
 elif platform.system() == 'Linux':  # pragma: no cover
-    lib_path = os.path.join(bin_path, 'libaudresample.so')
-elif platform.system() == 'Darwin':  # pragma: no cover
-    lib_path = os.path.join(bin_path, 'libaudresample.dylib')
+    lib_path = os.path.join(bin_path, 'linux', 'libaudresample.so')
+elif (
+        # MacOS Intel
+        platform.system() == 'Darwin'
+        and platform.processor() == 'i386'
+):  # pragma: no cover
+    lib_path = os.path.join(bin_path, 'macos-intel', 'libaudresample.dylib')
+elif (
+        # MacOS M1
+        platform.system() == 'Darwin'
+        and platform.processor() == 'arm'
+):  # pragma: no cover
+    lib_path = os.path.join(bin_path, 'macos-m1', 'libaudresample.dylib')
```

Comment:
The M1 dylib is not added, right?

Reply:
No, I just added it because I found out how it should look like ;)

To make it clearer, I removed the code and we propose to  add it when adding actual M1 support.

Reply:
Ah, so you will add it in a separate PR?

Reply:
I have now included all new binaries in this pull request, and also added the code for `mac0s-m1`.

Reply:
I also adjusted the description of the pull request to reflect better what I did.